Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs

```diff
@@ -617,28 +642,172 @@ public void Excluding_the_property_hiding_the_base_class_one_does_not_reveal_the
             act.Should().Throw<InvalidOperationException>().WithMessage("*No members were found *");
         }
 
-        public class BaseWithProperty
+        private class BaseWithProperty
         {
             public object Property { get; set; }
         }
 
-        public class SubclassAHidingProperty<T> : BaseWithProperty
+        private class SubclassAHidingProperty<T> : BaseWithProperty
         {
             public new T Property { get; set; }
         }
 
-        public class AnotherBaseWithProperty
+        private class BaseWithStringProperty
+        {
+            public string Property { get; set; }
+        }
+
+        private class SubclassHidingStringProperty : BaseWithStringProperty
+        {
+            public new string Property { get; set; }
+        }
+
+        private class AnotherBaseWithProperty
         {
             public object Property { get; set; }
         }
 
-        public class SubclassBHidingProperty<T> : AnotherBaseWithProperty
+        private class SubclassBHidingProperty<T> : AnotherBaseWithProperty
         {
             public new T Property
             {
                 get; set;
             }
         }
+
+        [Fact]
+        public void Ignores_fields_hidden_by_the_derived_class()
+        {
+            // Arrange
+            var subject = new SubclassAHidingField
+            {
+                Field = "DerivedValue"
+            };
+
+            ((BaseWithField)subject).Field = "ActualBaseValue";
+
+            var expectation = new SubclassBHidingField
+            {
+                Field = "DerivedValue"
+            };
+
+            ((AnotherBaseWithField)expectation).Field = "ExpectedBaseValue";
+
+            // Act / Assert
+            subject.Should().BeEquivalentTo(expectation, _ => _.IncludingFields());
```

Comment:
Should we keep this consistent with the majority of the test suite, i.e. using `opt => opt` or `options => options` instead of `_ => _`?

Reply:
Yeah, I'm fine with that. I only recently learned through Nuke that you can use `_` multiple times.

Reply:
It easily becomes hard to read for humans.

E.g. is this perfectly valid C# and the two `_` refers to different parameters.
```cs
Func<int, Func<int, int>> NestedUsageOfDiscards() => _ => _ => 42;
```

For more C# syntax horror, see https://www.youtube.com/watch?v=jaPk6Nt33KM 
or https://twitter.com/Lucas_Trz/status/1613644113814986767